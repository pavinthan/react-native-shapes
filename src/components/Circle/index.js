import Circle from './Circle';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Circle.propTypes = propTypes;
Circle.defaultProps = defaultProps;

export default Circle;
