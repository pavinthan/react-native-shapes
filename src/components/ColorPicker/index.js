import ColorPicker from './ColorPicker';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

ColorPicker.propTypes = propTypes;
ColorPicker.defaultProps = defaultProps;

export default ColorPicker;
