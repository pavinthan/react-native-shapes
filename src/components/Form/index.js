import Form from './Form';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Form.propTypes = propTypes;
Form.defaultProps = defaultProps;

export default Form;
