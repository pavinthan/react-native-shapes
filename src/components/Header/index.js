import Header from './Header';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Header.propTypes = propTypes;
Header.defaultProps = defaultProps;

export default Header;
