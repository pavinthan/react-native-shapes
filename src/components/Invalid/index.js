import Invalid from './Invalid';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Invalid.propTypes = propTypes;
Invalid.defaultProps = defaultProps;

export default Invalid;
