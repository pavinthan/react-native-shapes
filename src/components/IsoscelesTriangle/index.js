import IsoscelesTriangle from './IsoscelesTriangle';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

IsoscelesTriangle.propTypes = propTypes;
IsoscelesTriangle.defaultProps = defaultProps;

export default IsoscelesTriangle;
