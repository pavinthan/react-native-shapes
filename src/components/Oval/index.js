import Oval from './Oval';
import defaultProps from './defaultProps';
import propTypes from './propTypes';

Oval.propTypes = propTypes;
Oval.defaultProps = defaultProps;

export default Oval;
