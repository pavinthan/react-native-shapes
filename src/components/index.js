export { default as Circle } from './Circle';
export { default as ColorPicker } from './ColorPicker';
export { default as Form } from './Form';
export { default as Header } from './Header';
export { default as Invalid } from './Invalid';
export { default as IsoscelesTriangle } from './IsoscelesTriangle';
export { default as Oval } from './Oval';
export { default as Parallelogram } from './Parallelogram';
export { default as Polygon } from './Polygon';
export { default as Rectangle } from './Rectangle';
export { default as Whiteboard } from './Whiteboard';
