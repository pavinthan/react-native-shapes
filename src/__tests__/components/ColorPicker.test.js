import React from 'react';
import { ColorPicker } from '../../components';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const rendered = renderer.create(<ColorPicker />).toJSON();
  expect(rendered).toBeTruthy();
});

it('renders color picker component', () => {
  const tree = renderer.create(<ColorPicker />).toJSON();
  expect(tree).toMatchSnapshot();
});
