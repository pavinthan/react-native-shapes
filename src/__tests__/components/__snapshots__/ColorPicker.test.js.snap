// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders color picker component 1`] = `
Array [
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 17,
          "fontWeight": "700",
          "marginLeft": undefined,
          "paddingLeft": 4,
          "paddingTop": 1,
          "textAlign": "center",
        },
        Object {
          "color": "#000",
          "textAlign": "left",
        },
      ]
    }
    virtual={undefined}
  >
    Select Color
  </Text>,
  <View
    style={
      Array [
        Object {},
        Object {
          "flex": 1,
          "flexDirection": "row",
        },
      ]
    }
    virtual={undefined}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 0,
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#007aff",
                "fontSize": 20,
                "marginLeft": 16,
                "marginRight": 16,
                "paddingTop": 0,
              },
              Object {
                "marginLeft": 5,
              },
              Object {
                "color": "#000",
              },
            ],
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
        type="FontAwesome"
        virtual={undefined}
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 0,
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#007aff",
                "fontSize": 20,
                "marginLeft": 16,
                "marginRight": 16,
                "paddingTop": 0,
              },
              Object {
                "marginLeft": 5,
              },
              Object {
                "color": "#007aff",
              },
            ],
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
        type="FontAwesome"
        virtual={undefined}
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 0,
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#007aff",
                "fontSize": 20,
                "marginLeft": 16,
                "marginRight": 16,
                "paddingTop": 0,
              },
              Object {
                "marginLeft": 5,
              },
              Object {
                "color": "#5cb85c",
              },
            ],
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
        type="FontAwesome"
        virtual={undefined}
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 0,
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#007aff",
                "fontSize": 20,
                "marginLeft": 16,
                "marginRight": 16,
                "paddingTop": 0,
              },
              Object {
                "marginLeft": 5,
              },
              Object {
                "color": "#d9534f",
              },
            ],
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
        type="FontAwesome"
        virtual={undefined}
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "elevation": 0,
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#007aff",
                "fontSize": 20,
                "marginLeft": 16,
                "marginRight": 16,
                "paddingTop": 0,
              },
              Object {
                "marginLeft": 5,
              },
              Object {
                "color": "#f0ad4e",
              },
            ],
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
        type="FontAwesome"
        virtual={undefined}
      >
        
      </Text>
    </View>
  </View>,
]
`;
